app/services/common/base.rb

```diff
@@ -113,9 +113,7 @@ def batch_tag
 
       raise InvalidAttributes, "Expected barcode for batch '#{batch.arbitrary_id}' to be alphanumeric with 24 characters. Got: #{barcodes.join(', ')}" if matches.blank?
 
-      if matches&.size > 1
-        matches.sort! { |a, b| a <=> b }
-      end
+      matches.sort! { |a, b| a <=> b } if matches&.size > 1
```

Comment:
don't overoptimize... that `if` is not really needed. Just add a `&.sort {...}` at the end of line 112